loopout.go

```diff
@@ -86,12 +86,25 @@ func newLoopOutSwap(globalCtx context.Context, cfg *swapConfig,
 		return nil, fmt.Errorf("cannot initiate swap: %v", err)
 	}
 
+	swapInvoiceHash, err := swap.GetInvoiceHash(
+		cfg.lnd.ChainParams, swapResp.swapInvoice,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if swapInvoiceHash != swapHash {
```

Comment:
The purpose of the `validateLoopOutContract` function is to do this kind of validation. You can move this code into that function too. Added benefit is that the invoice only needs to be decoded once.

Reply:
Fixed in the latest commit. Moved the logic inside validateLoopOutContract and replaced GetInvoiceAmt and GetInvoiceHash with a more generic DecodeInvoice.

Question: I imagine DecodeInvoice already exists in lnd somewhere. Is there a point in duplicating the logic in loop?

Reply:
I am not aware of such a wrapper function in `lnd`. There the call is typically made to `zpay32` directly.